### PR TITLE
feat: added Login and signup pages,

### LINKED ISSUE #2: Feat "Authentication"
This issue is focused on:

- Creating Login and Signup Pages
- TopNav and SideNav components
- Adding Authentication feature in the code

### PR DESCRIPTION
**Pull request by vaibhav-mahalle**
Added following pages: 
- Login 

- SignUp

- TopNav

- SideNav

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  --black-color: hsl(220, 15%, 12%);
+  --yellow-light: #f7e7a1;
+  --primary-notes: #0f56b3;
+}
+
+.App {
+  font-family: Poppins;
+  min-width: 100vw;
+  min-height: 100vh;
+
```
Comment (nishasen) [inline src/App.css]: remove the extra space

File: src/frontend/component/TopNav/TopNav.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <header className="nav-container black-color">
+      <div className="flex-row gap p-l-3">
+        <AiOutlineMenu size={25} />
+        <Link to={"/"}>
+          <div className="txt-xlg">KipNote</div>
+        </Link>
+      </div>
+      {true ? (
+        <div className="p-r-1">
+          <Link to={"/login"}>
```
Comment (nishasen) [inline src/frontend/component/TopNav/TopNav.jsx]: `<Link to="/login">`
no need of the curly braces
```suggestion
          <Link to="/login">
```

File: src/frontend/component/TopNav/TopNav.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { Link, useNavigate } from "react-router-dom";
+import { BiLogIn, BiLogOut } from "react-icons/bi";
+
+export const TopNav = () => {
+  const navigate = useNavigate();
+  return (
+    <header className="nav-container black-color">
+      <div className="flex-row gap p-l-3">
+        <AiOutlineMenu size={25} />
+        <Link to={"/"}>
```
Comment (nishasen) [inline src/frontend/component/TopNav/TopNav.jsx]:
```suggestion
        <Link to="/">
```

File: src/frontend/pages/Auth/Auth.css [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    margin: 1rem;
+    padding: 1rem;
+    padding-top: 0.5rem;
+    box-shadow: 1px 2px 3px rgb(0 0 0 /20%);
+}
+
+.signupPage{
+    /*  display: flex; */
+    /* justify-content: center; */
+}
```
Comment (nishasen) [inline src/frontend/pages/Auth/Auth.css]: remove the commented code

File: src/frontend/pages/Auth/Signup.jsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            />
+          </div>
+
+          <button type="submit" className="btn btn-primary" >
+            Sign Up
+          </button>
+        </form>
+
+        <button className="btn btn-outline-secondary">
+          <Link to="/Login">Already a User? Login</Link>
```
Comment (nishasen) [inline src/frontend/pages/Auth/Signup.jsx]:
```suggestion
          <Link to="/login">Already a User? Login</Link>
```

**Review by nishasen — APPROVED** [review]
Suggested few changes, else everything looks good

**Review by shashidwivedi — APPROVED** [review]
Everything else looks good. Can work on changes suggested by Nisha

File: src/App.css (hunk 1 above)
Comment (vaibhav-mahalle) [reply src/App.css]: Thanks for pointing it out, I have made the changes in the latest commits.

File: src/frontend/component/TopNav/TopNav.jsx (hunk 3 above)
Comment (vaibhav-mahalle) [reply src/frontend/component/TopNav/TopNav.jsx]: Thanks for pointing it out, I have made the changes in the latest commits.

File: src/frontend/pages/Auth/Signup.jsx (hunk 5 above)
Comment (vaibhav-mahalle) [reply src/frontend/pages/Auth/Signup.jsx]: Thanks for pointing it out, I have made the changes in the latest commits.

File: src/frontend/component/TopNav/TopNav.css [hunk 6]
```diff
@@ -0,0 +1,14 @@
+.nav-container{
+    height: 4rem;
+    display: flex;
+    align-items: center;
+    justify-content: space-between ;
+    box-shadow: 0 3px 3px 2px rgb(0 0 0/20%);
```
Comment (aishwarya-bm) [inline src/frontend/component/TopNav/TopNav.css:6]: use color variable

File: src/frontend/pages/Auth/Auth.css [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    margin-bottom: 0.5rem;
+}
+
+.loginContainer{
+    display: flex;
+    flex-direction: column;
+    gap: 0.2rem;
+    padding: 2rem;
+    width: 22rem;
+    box-shadow: 1px 2px 3px rgb(0 0 0 /20%);
```
Comment (aishwarya-bm) [inline src/frontend/pages/Auth/Auth.css:48]: here also

**Review by aishwarya-bm — COMMENTED** [review]
Code looks clean and readable, liked the usage of clamp and navlinks.

**Pull request merged**